src/alchemlyb/estimators/ti_gaussian_quadrature_.py

```diff
@@ -0,0 +1,228 @@
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator
+from .. import concat
+from .base import _EstimatorMixOut
+
+
+class TI_GQ(BaseEstimator, _EstimatorMixOut):
+    """Thermodynamic integration (TI) with gaussian quadrature estimation.
+       When the simulations are performed at certain gaussian quadrature
+       points (lambdas), the free energy can be estimated using gaussian 
+       quadrature as an alternative to the trapezoidal rule.
+
+    Parameters
+    ----------
+
+    verbose : bool, optional
+        Set to True if verbose debug output is desired.
+
+    Attributes
+    ----------
+
+    delta_f_ : DataFrame
+        The estimated cumulative free energy from one state to another.
+
+    d_delta_f_ : DataFrame
+        The estimated statistical uncertainty (one standard deviation) in
+        dimensionless cumulative free energies.
+
+    states_ : list
+        Lambda states for which free energy estimation were obtained.
+
+    dhdl : DataFrame
+        The estimated dhdl of each state.
+
+
+    .. versionadded:: 2.0.0
+       `delta_f_`, `d_delta_f_`, `states_` are view of the original object.
+
+    """
+
+    def __init__(self, verbose=False):
+        self.verbose = verbose
+
+    def fit(self, dHdl):
+        """
+        Compute cumulative free energy from one state to another by integrating
+        dHdl across lambda values.
+
+        Parameters
+        ----------
+        dHdl : DataFrame
+            dHdl[n,k] is the potential energy gradient with respect to lambda
+            for each configuration n and lambda k.
+
+        """
+
+        # sort by state so that rows from same state are in contiguous blocks,
+        # and adjacent states are next to each other
+        dHdl = dHdl.sort_index(level=dHdl.index.names[1:])
+
+        # obtain the mean and variance of the mean for each state
+        # variance calculation assumes no correlation between points
+        # used to calculate mean
+        self.means = dHdl.groupby(level=dHdl.index.names[1:]).mean()
```

Comment:
Why do you need `self.means` and `self.variances`? I mean instead of just means and variances? Or even as private variable.